### PR TITLE
add migration slack_channel_id to employees table and refactor MessageSender#run

### PR DESCRIPTION
**Pull request by rjewell2200**
This PR fixes this issue, https://github.com/18F/dolores-landingham-bot/issues/74 -- thank you @noahschutte

### CODE REVIEW

File: app/services/message_sender.rb [hunk 1]
```diff
@@ -8,11 +8,16 @@ def initialize(employee, message)
 
   def run
     configure_slack
-    channel_id = SlackChannelIdFinder.new(employee.slack_username, client).run
 
-    if channel_id
+    if employee.slack_channel_id.nil?
```
Comment (jessieay) [inline app/services/message_sender.rb]: nice! this logic all looks good. can we write a test that verifies its behavior, though? something in `spec/services/message_sender_spec.rb` ? If you want pointers on that, I am happy to help!

Comment (rjewell2200) [inline app/services/message_sender.rb]: @jessieay no problem -- I just committed a change that adds a few tests to MessageSender. I also refactored the sweet to be more readable. Let me know what you think. Happy to adjust my work as needed.

**Comment by gemfarmer** [comment]
This is great! Thanks @rjewell2200 !

**Comment by rjewell2200** [comment]
Awesome! I saw there was one random failing test but I am certain that my
contribution didn't impact that failure. The same test fails in master.

On Mon, Mar 14, 2016 at 4:50 PM, Brian Hedberg <notifications@github.com>
wrote:

> This is great! Thanks @rjewell2200 <https://github.com/rjewell2200> !
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/18F/dolores-landingham-bot/pull/156#issuecomment-196516046>
> .
>



-- 

Robert Jewell
1.914.400.5219

File: spec/services/message_sender_spec.rb [hunk 2]
```diff
@@ -1,83 +1,95 @@
...
-      employee: employee,
-      scheduled_message: scheduled_message,
-      sent_on: Date.today,
-      sent_at: Time.parse("10:00:00 UTC"),
-      error_message: "",
-      message_body: scheduled_message.body,
-    )
-
-    Timecop.return
+  let(:base_dt) { Time.parse("10:00:00 UTC") }
```
Comment (jessieay) [inline spec/services/message_sender_spec.rb]: hey! do you mind including the setup for each test within the test itself -- similar to previous style? I generally try to stay away from `let` and `before` statements for these types of reasons: https://robots.thoughtbot.com/lets-not

**Comment by jessieay** [comment]
you're right - same failure on `master` -- wonder how that happened! will have to fix it up this Friday. Timezones are rough!

Once you update specs (remove `let` and `before` statements in favor of in-spec setup), this looks ready to :ship: !

**Comment by rjewell2200** [comment]
@jessieay I committed changes to the spec file as you requested. Let me know if the changes suit your style.

**Comment by jessieay** [comment]
@rjewell2200 thank you -- updates look great. Thank you for incorporating and adding some great tests!

Squashed and merged at 67629c784beab38dc4b914f16290fe9bf4e23aca

**Pull request closed without merging**